SONARJAVA-1221 Analyze test classes running with `Enclosed` runner

olcbean opened the pull request:
JUnit4 has introduced an `Enclosed` runner that runs all static inner
classes of the outer class.

This PR analyzes if there are tests in the inner classes to be run, and only
if there are none, `squid:S2187` will be triggered ( and report that there are
no tests in the outer class )

olcbean commented:
@AntonRiabov as I am not a core member of Sonar, I cannot make this call.
I would really like to further work on this PR, and get it merged, but so far there has been no communication from Sonar 😢

AntonRiabov commented:
> @AntonRiabov as I am not a core member of Sonar, I cannot make this call.
> I would really like to further work on this PR, and get it merged, but so far there has been no communication from Sonar 😢

@olcbean maybe you should write a comment to [this jira on sonar](https://jira.sonarsource.com/browse/SONARJAVA-1221) to reopen it, they closed this ticket, but feature not merged. I think they should reopen it to count it as TODO maybe then they will merge your pullrequest  to close ticket

olcbean commented:
@christophe-zurn-sonarsource thanks for your feedback! I just rebased and updated the PR accordingly. 

Can you take a look again when you have a chance ;)

benzonico commented:
merged via https://github.com/SonarSource/sonar-java/pull/2242 thanks for your contribution @olcbean
